### PR TITLE
Valid config

### PR DESCRIPTION
**Pull request by mattcollier**
Depends on https://github.com/w3c/web-ledger/pull/10

### CODE REVIEW

File: lib/LedgerEventStorage.js [hunk 1]
```diff
@@ -351,28 +350,31 @@ class LedgerEventStorage {
...
     };
     const projection = {_id: 0};
-    const records = await this.collection.find(query, projection).sort({
-      'meta.blockHeight': -1,
-      'meta.blockOrder': -1
-    }).limit(1).toArray();
+    const cursor = await this.collection.find(query, projection)
+      .sort({'meta.blockHeight': -1})
+      .hint('event.validConfiguration.core.1')
+      .limit(1);
```
[the remark this answers is not in the document]
Comment (mattcollier) [reply lib/LedgerEventStorage.js]: The rule is that there is only one validConfiguration per block.  Continuity only marks one configuration per block as being valid.  That seems like the best way to handle it to me.

[the remark this answers is not in the document]
Comment (mattcollier) [reply lib/LedgerEventStorage.js]: Perhaps `selectedConfiguration` would be better than `validConfiguration`?

[the remark this answers is not in the document]
Comment (dlongley) [reply lib/LedgerEventStorage.js]: @mattcollier -- `selectedConfiguration` would be better, but what about `activeConfiguration` instead? I like that better.

[the remark this answers is not in the document]
Comment (mattcollier) [reply lib/LedgerEventStorage.js]: Well, we're already using the `activeConfiguration` in reference to the configuration that was in effect at a given `blockHeight` as here:

https://github.com/digitalbazaar/bedrock-ledger-storage-mongodb/blob/ad368d018e732e368758401ff5dbfae6b0f99a04/lib/LedgerEventStorage.js#L361-L362

I suggest `effectiveConfiguration` as another alternative.

[the remark this answers is not in the document]
Comment (mattcollier) [reply lib/LedgerEventStorage.js]: And now that I think about it... I like `getEffectiveConfiguration` better for the function as well.

So using effective for both sounds good to me.  This is going to be a major release anyhow.  What say you?

[the remark this answers is not in the document]
Comment (dlongley) [reply lib/LedgerEventStorage.js]: I can live with either "active configuration" OR "effective configuration" everywhere. Just can't use both. Your pick.

**Comment by mattcollier** [comment]
@dlongley ready for another pass.

File: lib/LedgerEventStorage.js [hunk 2]
```diff
@@ -347,32 +346,35 @@ class LedgerEventStorage {
...
+      .sort({'meta.blockHeight': -1})
+      .hint('event.effectiveConfiguration.core.1')
+      .limit(1);
+    if(explain) {
+      return await cursor.explain('executionStats');
+    }
+    const records = await cursor.toArray();
     if(records.length === 0) {
       throw new BedrockError(
         'The active ledger configuration was not found.',
```
Comment (dlongley) [inline lib/LedgerEventStorage.js]: `active` => `effective`

**Pull request merged**